### PR TITLE
make yak-shave

### PR DESCRIPTION
**Pull request by kitcambridge**
* Add `html-server-cov`, `test-server`, and `bench-server` for testing the server without the smoke tests.
* Add some comments for the variables and recipes.
* Revert the `GOLDFLAGS` trick for setting the version via the linker.
* Install `go-capnproto` so that `capnpc` can use the `capnpc-go` binary.

@jrconlin r?

### CODE REVIEW

File: Godeps [hunk 1]
```diff
@@ -24,5 +23,6 @@ github.com/jacobsa/oglematchers 4fc24f97b5b74022c2a3f4ca7eed57ca29083d3e
 
 # Installable packages.
 github.com/gogo/protobuf/... d59ce9ecb817e6fbca932115f03520207f5a8a07
+github.com/glycerine/go-capnproto/... d56197a98e456a3b42c3bd9b25ff40a1a8f31420
 github.com/rafrombrc/gomock/... c922279faf77f29ce5781e96eb0711837fcb477c
```
Comment (jrconlin) [inline Godeps]: Double include? See line 20?

File: Makefile [hunk 2]
```diff
@@ -1,53 +1,75 @@
...
 $(BIN):
 	mkdir -p $(BIN)
 
+# Fetch and install dependencies.
 $(DEPS):
 	@echo "Installing dependencies"
-	GOPATH=$(GOPATH) $(GPM) install
-	GOPATH=$(GOPATH) $(GO) install github.com/gogo/protobuf/... \
+	$(GPM) install
+	$(GO) install github.com/gogo/protobuf/... \
```
Comment (jrconlin) [inline Makefile]: hrm. gpm should be able to install everything (you don't include the version if you don't want to vendor lock the library). Why are we calling gpm, then installing these libraries again?

File: Makefile [hunk 3]
```diff
@@ -1,53 +1,75 @@
...
-ifneq ($(strip $(VERSION)),)
-	GOLDFLAGS := -X $(PACKAGE)/simplepush.VERSION $(VERSION) $(GOLDFLAGS)
-endif
-
-
-.PHONY: all build clean test $(TARGET) memcached
+PATH := $(CURDIR)/bin:$(DEPS)/bin:$(PATH)
+GOPATH := $(DEPS):$(CURDIR)
+GPM := GOPATH=$(GOPATH) $(CURDIR)/gpm
+GO := GOPATH=$(GOPATH) go
```
Comment (jrconlin) [inline Makefile]: :+1:

File: Makefile (hunk 2 above)
Comment (kitcambridge) [inline Makefile]: gpm runs `go get -d`, which skips building the binaries. (Though it just occurred to me that gpm is checked into the repo, so we could just modify it directly...what do you think?)

Comment (jrconlin) [inline Makefile]: Ah, that makes sense. 

If we do modify gpm, we should do the right thing and submit a PR to that repo to optionally disable -d if a flag is specified. For now, having the second install is ok.

**Comment by jrconlin** [comment]
r+

**Pull request merged**